src/utils/helpers.ts

```diff
@@ -0,0 +1,20 @@
+import { Parameter } from 'src/Interfaces/Parameter';
+
+export function parameterArrayToMatrix(
+  arrayOfParameters: Array<Parameter>,
+  parametersPerSubArray: number,
+) {
+  const matrix: Array<Array<Parameter>> = [];
+  let i;
+  let k;
+
+  for (i = 0, k = -1; i < arrayOfParameters.length; i++) {
```

Comment:
`i` and `k` are essentially different indices and it is not apparent from their names. `i` is the array index, and `k` is the `matrixRowIndex`. You should name them something like that so it's easier to make sense of your code.

Reply:
Secondly, I think you can just declare them in-line like `for (let i = 0; let k = -1...`) - let me know if that is even possible and if you agree that to be the better way of writing

Reply:
for (let i = 0, k = -1; i < arrayToConvert.length; i++)

this is the correct syntax

Reply:
yeah, your syntax seems correct. I would still recommend renaming `i` and `k` to allow them to be more readable